### DISCUSSION TITLE
Export HTML Including Sidebar

### DISCUSSION THREAD

**Discussion by mhfowler** [outline/outline#5355] [Ideas]
It would be very nice to be able to take a snapshot of an outline and export it as HTML, including the sidebar. 

The current HTML export just includes the plain html pages, which is good for sort of saving the data, 
but not for saving the contents of an outline in a presentational way. 

Once a project has reached a point where an outline no longer needs to be edited, it makes more sense to save the files as a static HTML archive, instead of running the outline software, but to do this requires taking a complete snapshot of the outline. Are there tools that allow this? Or some way to do this already in outline?

**Comment by mhfowler** [comment]
tried to download all the pages of a collection using wget (with flags) but so far no luck, I'm guessing because of the javascript

**Reply by tommoor** [reply]
You would need to use a headless browser that can run JavaScript like puppeteer if you went this direction. I think the easiest would be to download the workspace as HTML export and then right a script to create an index page / sidebar based on the folder structure.

**Reply by mhfowler** [reply]
thats an interesting approach. might look into that. also considering trying to use gitbook to make a gitbook from the markdown export, as an alternative 

btw the html export also seems to be having a hard-time with a lot of characters (there might be some encoding issue)

for example here is how a page renders as html vs on live outline (and almost every page looks like this)

![Screenshot from 2023-05-18 11-20-00](https://github.com/outline/outline/assets/1831536/1474245c-49cd-442e-b8fc-9244dfa3bae6)

![Screenshot from 2023-05-18 11-19-50](https://github.com/outline/outline/assets/1831536/0e59c091-4bec-4b99-9d22-c40cdbdfb9d9)

**Reply by tommoor** [reply]
Could you upload the generated HTML file with these characters?

**Comment by mhfowler** [comment]
yup, here it is: https://nc.commoninternet.net/s/8NCKZGqDfR4tEC7

**Reply by tommoor** [reply]
The HTML looks and renders fine here. What OS and browser are you using?

**Reply by mhfowler** [reply]
hmm strange, I was using firefox on debian

**Comment by mhfowler** [comment]
aw figured out the issue was that I was using
`python3 -m http.server` to serve the html folder, and only when I do that do I see the strange characters. 

lame that python3 http.server does not properly handle them, 

but anyway sorry for the distraction and thinking it was a bug in outline. when I load a file directly in firefox it renders correctly. 

I ended up using mdbook to generate a website with a nice sidebar from the markdown files, which works pretty well.

**Reply by tommoor** [reply]
Seems like the http server does not use UT8 characterset by default https://devpress.csdn.net/python/62feedd2c677032930804a57.html
